src/smpi/bindings/smpi_pmpi_request.cpp

```diff
@@ -91,8 +145,14 @@ int PMPI_Start(MPI_Request * request)
   if (request == nullptr || *request == MPI_REQUEST_NULL) {
     retval = MPI_ERR_REQUEST;
   } else {
-    (*request)->start();
-    retval = MPI_SUCCESS;
+    if((*request)->flags() & SEND){
+      retval = persistent_send(request);
+    }else if((*request)->flags() & RECV){
+      retval = persistent_recv(request);
+    }else{
+      (*request)->start();
```

Comment:
I'm not sure if we need the else clause here. The standard says:

> A communication (send or receive) that uses  a  persistent  request  is initiated by the function MPI_Start.

Also check whether the request is a persistent one (see standard).